File: JSPM.Web/Modules/ServerTypes/Students/InstituteAcademicForm.ts
import { IntegerEditor, StringEditor, DecimalEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface InstituteAcademicForm {
    BranchId: IntegerEditor;
    AcademicYearId: IntegerEditor;
    SemesterId: IntegerEditor;
    DivisionId: IntegerEditor;
    Month: IntegerEditor;
    Year: IntegerEditor;
    EStatus: StringEditor;
    Grade: StringEditor;
    Percentage: DecimalEditor;
    StudentId: IntegerEditor;
}

export class InstituteAcademicForm extends PrefixedContext {
    static formKey = 'Students.InstituteAcademic';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!InstituteAcademicForm.init)  {
            InstituteAcademicForm.init = true;

            var w0 = IntegerEditor;
            var w1 = StringEditor;
            var w2 = DecimalEditor;

            initFormType(InstituteAcademicForm, [
                'BranchId', w0,
                'AcademicYearId', w0,
                'SemesterId', w0,
                'DivisionId', w0,
                'Month', w0,
                'Year', w0,
                'EStatus', w1,
                'Grade', w1,
                'Percentage', w2,
                'StudentId', w0
            ]);
        }
    }
}

File: JSPM.Web/Modules/Students/InstituteAcademic/InstituteAcademicForm.cs
using Serenity.ComponentModel;

namespace JSPM.Students.Forms;

[FormScript("Students.InstituteAcademic")]
[BasedOnRow(typeof(InstituteAcademicRow), CheckNames = true)]
public class InstituteAcademicForm
{
    public int BranchId { get; set; }
    public int AcademicYearId { get; set; }
    public int SemesterId { get; set; }
    public int DivisionId { get; set; }
    public int Month { get; set; }
    public int Year { get; set; }
    public string EStatus { get; set; }
    public string Grade { get; set; }
    public decimal Percentage { get; set; }
    public int StudentId { get; set; }
}